lib/ecto/adapters/postgres/connection.ex

```diff
@@ -71,7 +71,7 @@ if Code.ensure_loaded?(Postgrex.Connection) do
       where    = where(query.wheres, sources)
       group_by = group_by(query.group_bys, sources)
       having   = having(query.havings, sources)
-      order_by = order_by(query.order_bys, sources)
+      order_by = order_by(query.order_bys, query.distincts, sources)
```

Comment:
In master/0.9, distinct is singular. I would recommend you to write the patch against master instead. :)